merlin/core/compat.py

```diff
@@ -13,13 +13,22 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-HAS_GPU = False
 try:
     from numba import cuda
 
-    try:
-        HAS_GPU = len(cuda.gpus.lst) > 0
-    except cuda.cudadrv.error.CudaSupportError:
-        pass
 except ImportError:
     cuda = None
+
+HAS_GPU = False
+try:
+    import dask_cuda
+
+    HAS_GPU = dask_cuda.utils.get_gpu_count()
```

Comment:
One edge case in CPU-mode where you might have `dask_cuda` installed but are not running with nvml support (For example, running a merlin docker image without GPU config). This call raises `pynvml.nvml.NVMLError_LibraryNotFound`. Might be able to add an extra `except` for that.

Reply:
Thanks @oliverholworthy! I was about to ping you :)

Do you think it would be better to just try to import `pynvml` directly (rather than `dask_cuda`), and catch the `pynvml.nvml.NVMLError_LibraryNotFound` error?

Reply:
Yes, I think using `pynvml` instead of `dask_cuda` here would work. And then maybe don't need the `numba.cuda.gpus.lst` check if we do that.